File: 0283-move-zeroes/0283-move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int n = nums.length;
        int i =0 ,j = 0;
        int count = 0;
        while(i <n){
            if(nums[i] == 0){
                count++;
                i++;
                continue;
            }
            nums[i-count]  = nums[i];
            i++;
        }
        j = i-count;
        while(count>0){
            --count;
            nums[j++] = 0;
        }
    }
}

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        count = 0
        i=0
        for i in range(len(nums)):
            if nums[i] == 0:
                count+=1
            else:
                nums[i-count] = nums[i]
        nums[i-count+1:] = [0] * count
        return nums